lib/syskit/roby_app/plugin.rb

```diff
@@ -146,6 +156,53 @@ def self.setup(app)
 
                 rtt_core_model = app.default_loader.task_model_from_name("RTT::TaskContext")
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
+
+                # Log Transfer FTP Server spawned during Application#setup
+                setup_local_log_transfer_server(app) if log_transfer_ip
+            end
+
+            def self.setup_local_log_transfer_server(app)
+                @tmp_root_ca = TmpRootCA.new(log_transfer_ip)
+                start_local_log_transfer_server(app.log_dir, @tmp_root_ca)
+                config_log_transfer
+            end
+
+            def self.send_file_transfer_command(name, logfile)
+                # Establishes communication with said process server
+                client = Syskit.conf.process_server_for(name)
+
+                # Commands method log_upload_file from said process server
+                client.log_upload_file(
+                    @log_transfer_ip,
+                    @log_transfer_port,
+                    @log_transfer_certificate,
+                    @log_transfer_user,
+                    @log_transfer_password,
+                    logfile
+                )
+                client
+            end
+
+            def self.config_log_transfer
+                @log_transfer_port = @log_transfer_server.port
+                @log_transfer_certificate = @tmp_root_ca.certificate
+                @log_transfer_user = @log_transfer_server.user
+                @log_transfer_password = @log_transfer_server.password
+            end
```

Comment:
Instead of copying the value into separate attributes, just create separate accessor methods

```
def log_transfer_port
    @log_transfer_server.port
end
```